aten/src/ATen/core/jit_type.h

```diff
@@ -1850,7 +1851,8 @@ struct CAFFE2_API InterfaceType : public NamedType {
 
   bool operator==(const Type& rhs) const override {
     if (auto user_rhs = rhs.cast<InterfaceType>()) {
-      return name() == user_rhs->name();
+      return isSubTypeImpl(*this, *user_rhs, nullptr) &&
```

Comment:
If an interface has a method which takes the interface itself as an argument, does this infinitely recurse? I feel like it would, but I also feel like the previous implementation would do that as well.